### PR TITLE
ctop: Submission of new port (v0.7.5)

### PR DESCRIPTION
**Pull request by judaew**
###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 11.2.3
Xcode 12.4

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

File: sysutils/ctop/Portfile [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    file copy ${distpath}/${distfiles} ${worksrcpath}/${name}
+}
+
+checksums           rmd160  f81e9814dcf096d4293a27bc42fca78a829a21a1 \
+                    sha256  bb7e51429ff03a2c065873c3bd9e7448af1b9733a4b58b71de12bc4a728879de \
+                    size    11529016
+
+use_configure       no
+
+build {}
```
Comment (herbygillot) [inline sysutils/ctop/Portfile:37]: `ctop` should be buildable from source using the `golang` portgroup.

```
build.cmd      make
build.pre_args VERSION=${version}
build.args     build
```

You can choose to vendor dependencies using the `go2port` tool, but if that does not work, you can remove `GO111MODULE=off` and `GOPROXY=off` from `build.env`.

Comment (herbygillot) [reply sysutils/ctop/Portfile:37]: Building it from source also means that you can support any arch that MacPorts' Go will build on, allowing you to remove `supported_archs x86_64`.

**Comment by judaew** [comment]
Done

**Comment by herbygillot** [comment]
Something broke in MacPorts CI, builds are failing across multiple pull requests.

**Comment by herbygillot** [comment]
Closing and reopening to rerun CI.

**Comment by herbygillot** [comment]
OK, looks good.  The only nitpick is since you're not using the Makefile as I suggested in my first comment, `ctop` is getting built without `-ldflags` being set, meaning there is no version information being compiled into the executable.

So if someone does `ctop -v`, they get this:

```
❯ ./ctop -v
ctop version dev-build, build none go1.16.3
```

...where one would expect to see something like "v0.7.5" instead of _'dev-build'_.

**Comment by judaew** [comment]
Sorry, I missed your suggestion about the Makefile. I like this option better.

File: sysutils/ctop/Portfile [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    multiple containers.\nctop comes with built-in support for Docker and \
+    runC\; connectors for other container and cluster systems are planned for \
+    future releases.
+homepage            https://ctop.sh
+
+checksums           rmd160  8743fb5fd970399665c5ecc9ec885635af495e11 \
+                    sha256  b191ffa8ba343edb47644fcbc57868a282a4bd720b1836f8e0ec15820880d23d \
+                    size    1291305
+
+set date_build [clock format [clock seconds] -format %Y-%m-%d]
```
Comment (judaew) [inline sysutils/ctop/Portfile:25]: If I am not mistaken, then the BUILD from the Makefile indicates the build date. What do you think of this way of specifying the date?

File: sysutils/ctop/Portfile [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+# FIXME: This port currently can't be built without allowing go mod to fetch
+build.env-delete    GOPROXY=off GO111MODULE=off
+
+build.cmd           make
+build.pre_args      VERSION=${version} BUILD=${date_build}
```
Comment (judaew) [inline sysutils/ctop/Portfile:31]: If I am not mistaken, then the BUILD from the Makefile indicates the build date. What do you think of this way of specifying the date?

Comment (herbygillot) [reply sysutils/ctop/Portfile:31]: This looks 100% fine to me

**Comment by herbygillot** [comment]
Thank you @judaew

**Comment by herbygillot** [comment]
One last comment is that the commit message for new ports should look something like:

```
ctop: new port
```

or

```
ctop: submission
```

...or at least that's the standard format.

**Comment by judaew** [comment]
Thanks, I will be know it 🌟

**Pull request merged**